portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -75,26 +78,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // comment thread, visible by all
     out.println("<ul>");
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
     for (Entity entity : results.asIterable()) {
       String comment = (String) entity.getProperty("comment");
       long timestamp = (long) entity.getProperty("timestamp");
-      String id = (String) entity.getProperty("id");
-      String nickname = getUserNickname(id);
-    
-      out.println("<li><span style='color:blue;'>" + convertTime(timestamp) + 
-                  "</span> <span style='color:green'>" + nickname + "</span>: " + 
-                  comment + "<form method='POST' action='/delete'>" +
-                                "<input type='submit' name='deleteBtn' value='Delete' />" +
-                                "<input type='hidden' name='comment' value='" + id + "'/>" +
-                            "</form> </li>");
+      Key key_id = entity.getKey();
+      String comment_id = Long.toString((long) entity.getProperty("comment_id"));
+      String user_id = (String) entity.getProperty("user_id");
+      double sentiment = (double) entity.getProperty("sentiment");
+      String sentimentHex = getColor(sentiment);
+      String nickname = getUserNickname(user_id);
+
+      out.println("<li>" +
+                    "<span style='color:blue;'>" + convertTime(timestamp) + "</span> " +
+                    "<span style='color: black;'>" + nickname + "</span>: " +
+                    "<span style='color: " + sentimentHex + ";'>" + comment + "</span>" +
+                    "<form method='POST' action='/delete'>"+
+                        "<input type='submit' name='deleteBtn' value='Delete'/>" +
+                        "<input type='hidden' name='comment' value='" + comment_id + "'/>" +
+                        "</form> </li>");
     } 
     out.println("</ul>");
     
   }
 
+  /**
+   * Convert a sentiment value in the range [-1, 1] to a color
+   * between red and green to reflect sentiment 
+  */
+  private String getColor(double sentiment) {
```

Comment:
It is better to point out what is the range of the input sentiment in the comment, and it would also help to handle the corner cases.

Reply:
I just updated my comments, do they now address this?